### PR TITLE
feat(cc-digital-channel): add-uts-for-digital-channel

### PR DESCRIPTION
**Pull request by akulakum**
# COMPLETES #https://jira-eng-sjc12.cisco.com/jira/browse/CAI-7458

## This pull request addresses

Adding comprehensive unit tests for the cc-digital-channels package to improve code coverage and test reliability.

## by making the following changes

- Added unit tests for useDigitalChannelsInit and useDigitalChannelsData hooks covering initialization, error handling, and edge cases.
- Created dedicated DigitalChannelsComponent.test.tsx for presentational component tests (theme rendering, props validation).
- Added ErrorBoundary integration test for DigitalChannels component

Screenshot 2026-02-05 at 3 44 04 PM

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

- [ ] The testing is done with the amplify link
< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

## The GAI Coding Policy And Copyright Annotation Best Practices ##

- [ ] GAI was not used (or, no additional notation is required)
- [ ] Code was generated entirely by GAI
- [ ] GAI was used to create a draft that was subsequently customized or modified
- [ ] Coder created a draft manually that was non-substantively modified by GAI (e.g., refactoring was performed by GAI on manually written code)
- [x] Tool used for AI assistance (GitHub Copilot / Other - specify)
  - [ ] Github Copilot
  - [ ] Other - Please Specify
- [ ] This PR is related to
  - [ ] Feature
  - [ ] Defect fix
  - [x] Tech Debt
  - [ ] Automation

### Checklist before merging

- [ ] I have not skipped any automated checks
- [x] All existing and new tests passed
- [ ] I have updated the testing document
- [x] I have tested the functionality with amplify link

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx [hunk 1]
```diff
@@ -66,68 +84,147 @@ import {DigitalChannels} from '../../src/digital-channels';
...
     expect(engageWidget).toBeInTheDocument();
+    expect(engageWidget).toHaveAttribute('data-theme', 'dark');
   });
 
   it('should have proper store integration', async () => {
     // eslint-disable-next-line @typescript-eslint/no-require-imports
     const storeModule = require('@webex/cc-store');
     expect(storeModule.default.currentTask).toBeTruthy();
     expect(storeModule.default.logger).toBeTruthy();
+    expect(storeModule.default.dataCenter).toBe('produs1');
```
Comment (adhmenon) [inline packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx]: Do we know this is always going to be the default?

**Review by adhmenon — APPROVED** [review]
LGTM. One question, only.

File: packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx (hunk 1 above)
Comment (akulakum) [reply packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx]: Good catch!  'produs1' is not a default value - it's a mock test value we configure in the test setup. I've refactored to use a MOCK_DATA_CENTER constant to make this clearer. The test verifies that the component correctly receives whatever dataCenter value is configured in the store.

File: packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx [hunk 2]
```diff
@@ -140,10 +249,147 @@ describe('DigitalChannels Component - Integration Tests with Real Components', (
...
-      expect(container.querySelector('md-theme')).not.toBeNull();
+      expect(screen.container.querySelector('md-theme')).toBeNull();
     });
+
+    // Verify no DOM elements are rendered when dataCenter is empty
+    expect(screen.container.querySelector('md-theme#app-theme')).toBeNull();
+    expect(screen.container.querySelector('div > md-theme')).toBeNull();
+    expect(screen.queryByTestId('engage-widget')).toBeNull();
+    expect(screen.container.innerHTML).toBe('');
+
```
Comment (rsarika) [inline packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx:299]: **Suggestion (non-blocking):** Consider extracting the store state restoration logic into a nested `describe` block with `afterEach` for cleaner test isolation:

```typescript
describe('when dataCenter is empty', () => {
  let originalDataCenter: string;
  
  beforeEach(() => {
    originalDataCenter = store.dataCenter;
    act(() => { runInAction(() => { store.dataCenter = ''; }); });
  });
  
  afterEach(() => {
    act(() => { runInAction(() => { store.dataCenter = originalDataCenter; }); });
  });
  
  it('should not render', async () => { /* ... */ });
});
```

This pattern makes test cleanup more explicit and less error-prone.

File: packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx [hunk 3]
```diff
@@ -140,10 +249,147 @@ describe('DigitalChannels Component - Integration Tests with Real Components', (
...
+  it('should handle error gracefully when onErrorCallback is undefined', async () => {
+    // eslint-disable-next-line @typescript-eslint/no-require-imports
+    const storeModule = require('@webex/cc-store');
+    const store = storeModule.default;
+    const originalCallback = store.onErrorCallback;
+    store.onErrorCallback = undefined;
+
+    const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+    mockShouldThrow = true;
+
```
Comment (rsarika) [inline packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx:393]: **Suggestion (non-blocking):** The `consoleErrorSpy` pattern is used multiple times. Consider extracting to a reusable helper or using `beforeEach`/`afterEach`:

```typescript
let consoleErrorSpy: jest.SpyInstance;

beforeEach(() => {
  consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
});

afterEach(() => {
  consoleErrorSpy.mockRestore();
});
```

File: packages/contact-center/cc-digital-channels/tests/helper.ts [hunk 4]
```diff
@@ -95,4 +94,132 @@ describe('useDigitalChannelsInit', () => {
...
+    );
+  });
+
+  it('should not throw when logger is undefined', async () => {
+    const getAccessToken = jest.fn().mockRejectedValue(new Error('token error'));
+
+    const {result} = renderHook(() =>
+      useDigitalChannelsData({
+        getAccessToken,
+        currentTask: defaultTask,
```
Comment (rsarika) [inline packages/contact-center/cc-digital-channels/tests/helper.ts:217]: **Suggestion (non-blocking):** Consider a more descriptive test name that captures the full scenario:

```typescript
it('should handle token fetch error gracefully when logger is undefined')
```

This makes test failures easier to diagnose.

File: packages/contact-center/store/tests/storeEventsWrapper.ts [hunk 5]
```diff
@@ -916,6 +933,56 @@ describe('storeEventsWrapper', () => {
...
+        jest.clearAllMocks();
+      });
+
+      it('should return the access token on success', async () => {
+        const mockAccessToken = 'mock-access-token-12345';
+        // @ts-expect-error - webex credentials API not typed on IContactCenter
+        storeWrapper['store'].cc.webex.credentials.getUserToken = jest
+          .fn()
+          .mockResolvedValue({access_token: mockAccessToken});
+
```
Comment (rsarika) [inline packages/contact-center/store/tests/storeEventsWrapper.ts:948]: **Suggestion (non-blocking):** There are several `@ts-expect-error` comments in these tests for the webex credentials API. For future maintainability, consider:

1. Creating a typed mock interface for the credentials API, or
2. Adding type definitions to the store's type declarations

This would improve IDE support and catch potential type mismatches early.

**Review by rsarika — APPROVED** [review]
Great work on adding comprehensive unit tests for the cc-digital-channels package! The tests are well-structured, follow proper MobX/React testing patterns, and cover important edge cases including error handling.

**Highlights:**
- Good use of `act()` with `runInAction()` for MobX state updates
- Proper mock strategy using `jest.requireActual`
- Nice refactor to use named constants for mock values
- Good coverage of error scenarios

Left a few minor suggestions as inline comments - these are non-blocking improvements for future consideration.

File: packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx (hunk 2 above)
Comment (akulakum) [reply packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx:299]: Thanks for the suggestion! Refactored to use nested describe blocks with beforeEach/afterEach for dataCenter and currentTask tests. This makes the test cleanup more explicit and isolated.

File: packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx (hunk 3 above)
Comment (akulakum) [reply packages/contact-center/cc-digital-channels/tests/digital-channels/index.tsx:393]: Good catch! Moved consoleErrorSpy to beforeEach/afterEach in the ErrorBoundary describe block. Also reset mockShouldThrow in afterEach for cleaner test isolation.

File: packages/contact-center/cc-digital-channels/tests/helper.ts (hunk 4 above)
Comment (akulakum) [reply packages/contact-center/cc-digital-channels/tests/helper.ts:217]: Updated the test name to 'should handle token fetch error gracefully when logger is undefined' for better clarity on what scenario is being tested.

File: packages/contact-center/store/tests/storeEventsWrapper.ts (hunk 5 above)
Comment (akulakum) [reply packages/contact-center/store/tests/storeEventsWrapper.ts:948]: This is a larger refactoring effort for future consideration - no changes made as it's non-blocking.
